### PR TITLE
migrations: Consolidate migrator docs

### PR DESCRIPTION
**Pull request by efritz**
This PR moves the docs that mention how to invoke the `migrator` into a central place under admin / how-to. This allows us to simply reference "run _this particular command_ in _your_ migrator" for future docs.

Let me know if this refactor has mad anything less clear, or if new docs need additional context or clarity.

## Test plan

I used my eyes: they're docs.

### CODE REVIEW

**Comment by efritz** [comment]
Will wait until this lands to resume work on:

- https://github.com/sourcegraph/sourcegraph/pull/30776
- https://github.com/sourcegraph/sourcegraph/pull/30760

...which would greatly benefit from this.

File: doc/admin/how-to/run_migrator_operations.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
+# How to run `migrator` operations
+
+> NOTE: This feature is only available in versions `3.37` and later
+
+The `migrator` is a service that runs as an initial step of the upgrade process for [Kubernetes](http://localhost:5080/admin/install/kubernetes/update#database-migrations) and [Docker-compose](/admin/install/docker-compose/operations#database-migrations) instance deployments. This service is also designed to be invokable directly by a site administrator to perform common tasks dealing with database state.
```
[the remark this answers is not in the document]
Comment (efritz) [reply doc/admin/how-to/run_migrator_operations.md]: Fixed the links.

File: doc/admin/how-to/index.md [hunk 2]
```diff
@@ -1,5 +1,6 @@
 # How-to guides
 
+- [How to run `migrator` operations](run_migrator_operations.md)
```
[the remark this answers is not in the document]
Comment (efritz) [reply doc/admin/how-to/index.md]: Updated all links and filename.

File: doc/admin/install/docker-compose/operations.md [hunk 3]
```diff
@@ -188,128 +188,13 @@ You can see what's changed in the [Sourcegraph changelog](../../../CHANGELOG.md)
...
+To execute the database migrations independently, follow the [docker-compose instructions on how to run a `migrator` command](/admin/how-to/run_migrator_operations#docker-compose). Running the `up` (default) command on the `migrator` of the *version you are upgrading to* will apply all migrations required by the next version of Sourcegraph.
+
+> NOTE: These instructions assume that the environment has all three databases enabled. If the configuration flag `DISABLE_CODE_INSIGHTS` is set and the `codeinsights-db` is unavailable, the operation will fail. To work around this, explicitly supply database(s) via the `-db` flag (e.g., `-db=frontend,codeintel`) for all commands.
```
[the remark this answers is not in the document]
Comment (efritz) [reply doc/admin/install/docker-compose/operations.md]: Updated.

File: doc/admin/how-to/index.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (efritz) [reply doc/admin/how-to/index.md]: Updated the structure of the doc as much as I could as well and gave explicit links from the index page.

**Review by danieldides — APPROVED** [review]
🎉

**Pull request merged**